torrent.go

```diff
@@ -1200,6 +1204,9 @@ func (t *Torrent) deleteConnection(c *PeerConn) (ret bool) {
 	}
 	_, ret = t.conns[c]
 	delete(t.conns, c)
+	if !t.cl.config.DisablePEX {
+		t.pex.Drop(c)
```

Comment:
Perhaps pex could be an interface, and a dummy provided when PEX is disabled. That way we're not testing everywhere whether we should be calling methods on it.

Reply:
It could, yes, but without a good deal of an alternative behaviour this doesn't feel like a big win. Also, we could have these checks internally in the methods but there are already quite a bit of things going on already.  I think what we've got here is a good balance on keeping clutter out.